base_report_to_printer/static/src/js/qweb_action_manager.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('base_report_to_printer.print', function (require) {
+    'use strict';
+
+    var ActionManager = require('web.ActionManager');
+    var core = require('web.core');
+    var framework = require('web.framework');
+    var rpc = require('web.rpc');
+
+
+    ActionManager.include({
+        ir_actions_report: function (action, options) {
+            var action_val = _.clone(action);
+            var _t = core._t;
+            var self = this;
+            var _super = this._super;
+
+            if ('report_type' in action_val && action_val.report_type === 'qweb-pdf') {
+                framework.blockUI();
+                rpc.query({
+                    model: 'ir.actions.report',
+                    method: 'print_action_for_report_name',
+                    args: [action_val.report_name]
+                }).then(function (print_action) {
+                    if (print_action && print_action.action_val === 'server') {
+                        framework.unblockUI();
+                        rpc.query({
+                            model: 'ir.actions.report',
+                            method: 'print_document',
+                            args: [action_val.context.active_ids, action_val.report_name],
+                            kwargs: {data: action_val.data || {}},
+                            context: action_val.context || {}
+                        }).then(function () {
+                            self.do_notify(_t('Report'),
```

Comment:
@lasley  Essentially I just want to replicate this block in python - so the top right kind of notification

Reply:
Damn I thought that was the case - I don't know of a way to do this. The [crash manager](https://github.com/odoo/odoo/blob/cd48a806ac636806130efcf5c055f73758cb0902/addons/web/static/src/js/services/crash_manager.js) is what handles anything we can send back error-wise from Python, and it does dialogs.

Reply:
An interesting module idea would be a `NotificationError` in Python and something in JS that intercepts the internal RPC calls, catches errors, and handles appropriately if a notification error. I don't think that exists though.